src/clients/ConfigStoreClient.ts

```diff
@@ -85,6 +95,17 @@ export class AcrossConfigStoreClient {
     return config.transferThreshold;
   }
 
+  getSpokeTargetBalancesForBlock(
+    l1Token: string,
+    blockNumber: number = Number.MAX_SAFE_INTEGER
+  ): { [chainId: number]: SpokePoolTargetBalance } {
+    const config = (sortEventsDescending(this.cumulativeTokenTransferUpdates) as SpokeTargetBalanceUpdate[]).find(
+      (config) => config.blockNumber <= blockNumber && config.l1Token === l1Token
```

Comment:
```suggestion
      (config) => config.l1Token === l1Token && config.blockNumber <= blockNumber
```

tiny nit but small optimization to filter on l1 token first. I doubt there will be 1000s of this event but just good practice to filter on rarer conditions first

Reply:
Good point!